fix: TestIntegrationTestSuite flaky due to TempDir cleanup race

## Summary

`TestIntegrationTestSuite` in `test/util/testnode` is flaky due to a race condition between Go's `t.TempDir()` cleanup and background goroutines (gRPC servers) that are still running when the test tears down.

## Error

```
--- FAIL: TestIntegrationTestSuite (9.08s)
    network.go:97: tearing down testnode
    testing.go:1369: TempDir RemoveAll cleanup: unlinkat /tmp/TestIntegrationTestSuite3052499985/001/testnode/data: directory not empty
```

The gRPC servers are still starting up after teardown:
```
failed to start gRPC server err=null
```

## Observed in

- Merge queue CI run for PR #6724: https://github.com/celestiaorg/celestia-app/actions/runs/22732380741

## Suggested fix

Ensure all background goroutines (particularly gRPC servers) are properly shut down before test teardown completes. Options:
1. Use `t.Cleanup()` to explicitly stop gRPC servers before TempDir cleanup
2. Replace `t.TempDir()` with a manually managed temp directory that tolerates non-empty removal
3. Add a sync mechanism to wait for all goroutines to stop during teardown

--- a/test/util/testnode/network.go
+++ b/test/util/testnode/network.go
@@ -75,6 +75,7 @@ func tryStartNetwork(t testing.TB, config *Config) (cctx Context, rpcAddr, grpcA
 	// Build up cleanup incrementally so that if a later step fails, the
 	// returned cleanup function tears down all previously acquired resources.
 	cleanup = func() {
+		t.Log("tearing down testnode")
 		cancel()
 	}
 
@@ -82,8 +83,9 @@ func tryStartNetwork(t testing.TB, config *Config) (cctx Context, rpcAddr, grpcA
 	if err != nil {
 		return Context{}, "", "", cleanup, err
 	}
+	cleanupCancel := cleanup
 	cleanup = func() {
-		cancel()
+		cleanupCancel()
 		if err := stopNode(); err != nil {
 			t.Logf("error stopping node %v", err)
 		}
@@ -98,9 +100,9 @@ func tryStartNetwork(t testing.TB, config *Config) (cctx Context, rpcAddr, grpcA
 	if err != nil {
 		return Context{}, "", "", cleanup, err
 	}
-	prevCleanup := cleanup
+	cleanupNode := cleanup
 	cleanup = func() {
-		prevCleanup()
+		cleanupNode()
 		if err := cleanupGRPC(); err != nil {
 			t.Logf("error when cleaning up GRPC %v", err)
 		}
@@ -110,10 +112,9 @@ func tryStartNetwork(t testing.TB, config *Config) (cctx Context, rpcAddr, grpcA
 	if err != nil {
 		return Context{}, "", "", cleanup, err
 	}
-	prevCleanup2 := cleanup
+	cleanupGRPCServer := cleanup
 	cleanup = func() {
-		t.Log("tearing down testnode")
-		prevCleanup2()
+		cleanupGRPCServer()
 		if err := apiServer.Close(); err != nil {
 			t.Logf("error when closing API server %v", err)
 		}

